### ISSUE TITLE
[Problem] macOS: PartDesign Link window moves to behind FreeCAD window when clicking FreeCAD window

### ISSUE THREAD

**Issue by Anthony-Gaudino** [FreeCAD/FreeCAD#10840]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

In PartDesign some features, like Pad, have a `Property -> Profile` property with a three dots `...` (one dot on 0.21.1).
Clicking on this button a `Link` window opens.
<img width="552" alt="Screenshot 2023-09-27 at 10 24 40" src="https://github.com/FreeCAD/FreeCAD/assets/2439728/a75ffb06-8637-486f-998a-ec9dee62f670">
<img width="436" alt="Screenshot 2023-09-27 at 10 25 24" src="https://github.com/FreeCAD/FreeCAD/assets/2439728/ec2b0098-5b4f-4958-bac1-c374897ebfd9">

The issue with this window is that clicking anywhere on the main FreeCAD window will move the Link window to behind it, so I can't access it. To access the Link window I need to move the main FreeCAD window.

The Link window should always stay on top of the main FreeCAD window to avoid this problem.

### Full version info

```shell
OS: macOS 14.5
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38079 (Git)
Build type: Release
Branch: main
Hash: de508e437d13e7eb35baffceea496fb9107888cb
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C)
Installed mods: 
  * Launcher
  * Nodes 0.1.36
  * InventorLoader 1.4.0
  * Assembly4.backup1719958896.85335 0.50.12 (Disabled)
  * Render.backup1719959316.120863 2024.6.5 (Disabled)
  * woodworking 0.21.2.33771
  * ThreadProfile.backup1719959316.6397839 1.90.0 (Disabled)
  * MeshRemodel.backup1719959315.871541 1.9.23 (Disabled)
  * Silk 0.1.5
  * Defeaturing 1.2.2
  * Pyramids-and-Polyhedrons
  * Curves.backup1719958901.528519 0.6.36 (Disabled)
  * fasteners.backup1719958901.808955 0.5.20 (Disabled)
  * nurbs
  * sheetmetal.backup1719959316.3711638 0.4.21 (Disabled)
  * A2plus.backup1719958896.309429 0.4.64a (Disabled)
  * Manipulator 1.5.7
  * lattice2 1.0.0
  * CurvedShapes 1.0.9
  * toSketch 1.0.1
  * Curves 0.6.40
  * Assembly4 0.50.15
  * Assembly3 0.12.2
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Anthony-Gaudino** [comment]
> Works OK here:
> 
> ```
> OS: Windows 8 build 9600
> Word size of FreeCAD: 64-bit
> Version: 0.21.1.33668 +26 (Git)
> Build type: Release
> Branch: (HEAD detached at 0.21.1)
> Hash: f6708547a9bb3f71a4aaade12109f511a72c207c
> Python 3.8.10, Qt 5.15.2, Coin 4.0.1, Vtk 8.2.0, OCC 7.6.3
> Locale: C/Default (C) [ OS: Dutch/Netherlands (nl_NL) ]
> Installed mods:
> ```

Probably this is MacOS specific. This problem also existed on 0.20.2.

**Comment by luzpaz** [comment]
@PLChris can you verify on macOS ?

**Comment by luzpaz** [comment]
@JohnOCFII can you reproduce on latest macOS dev build ?

**Comment by JohnOCFII** [comment]
Yes, problem still exists in current weekly build.  It took me a while to find how to access the `Link` window, as it is the single`.` on the right of the Profile name that is selected to access it.  Hard to see!  

If this is a "regular" feature, and not some under-the-hood developer debugging tool, I'd wish for a more visible access method (in addition to keeping the Links window frontmost in the stack of FreeCAD windows).

<img width="451" alt="Screenshot 2023-11-15 at 12 12 13 PM" src="https://github.com/FreeCAD/FreeCAD/assets/7844212/1da42eb8-bda6-4506-8b00-e6f6901e845f">

```
OS: macOS 14.1.1
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.35024 (Git)
Build type: Release
Branch: main
Hash: 4ff00ef99869e35295a5a17b8f486901504c655b
Python 3.10.13, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.6, OCC 7.6.3
Locale: C/Default (C)
```

**Comment by luzpaz** [comment]
@adrianinsaval is this a UI tweek you could offer ? Not sure how it would be made just for macOS though...any idea ?

**Comment by Anthony-Gaudino** [comment]
Updated Full version info as it still happens on 0.22.0dev.

Also, screen recording:

https://github.com/user-attachments/assets/95cd1cea-8179-4941-9a33-cd0b36ae5bef

**Comment by FEA-eng** [comment]
@maxwxyz Can you check if it's still relevant on Mac?

**Comment by maxwxyz** [comment]
Still relevant and very annoying. Happens with other windows too.
